### PR TITLE
Bug 1821005 - Restore previous Gradle 8 upgrade reverted due to startup crashes

### PR DESCRIPTION
**Pull request by mcarare**
This reverts commit 592e88b3547d2e96ae04d8b5bd69cbab2ac3e97e.

### Pull Request checklist

- [ ] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/firefox-android/blob/main/docs/changelog.md) or does not need one
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/firefox-android/blob/main/docs/shared/android/accessibility_guide.md) or does not include any user facing features

### After merge
- **Breaking Changes**: If this is a breaking Android Components change, please push a draft PR on [Reference Browser](https://github.com/mozilla-mobile/reference-browser) to address the breaking issues.

### To download an APK when reviewing a PR (after all CI tasks finished running):
1. Click on `Checks` at the top of the PR page.
2. Click on the `firefoxci-taskcluster` group on the left to expand all tasks.
3. Click on the `build-apk-{fenix,focus,klar}-debug` task you're interested in.
4. Click on `View task in Taskcluster` in the new `DETAILS` section.
5. The APK links should be on the right side of the screen, named for each CPU architecture.

### GitHub Automation
https://bugzilla.mozilla.org/show_bug.cgi?id=1821005

### CODE REVIEW

File: focus-android/gradle.properties [hunk 1]
```diff
@@ -25,3 +25,5 @@ libUrl=https://github.com/mozilla-mobile/firefox-android/tree/main/focus-android
 libVcsUrl=https://github.com/mozilla-mobile/firefox-android.git
 libLicense=MPL-2.0
 libLicenseUrl=https://www.mozilla.org/en-US/MPL/2.0/
+android.nonTransitiveRClass=false
+android.nonFinalResIds=false
```
Comment (rahulsainani) [inline focus-android/gradle.properties:29]: Are we enabling r8 full mode in focus?

Comment (mcarare) [reply focus-android/gradle.properties:29]: No, I tested and minifying with R8 full mode also crashes Focus. 
As for the sample apps they do not have minify enabled.

Comment (rahulsainani) [reply focus-android/gradle.properties:29]: Yes, that's what my understanding was. I see we've now added the property below 👍

Comment (mcarare) [reply focus-android/gradle.properties:29]: Yes, thanks for pointing that out!

**Review by jonalmeida — APPROVED** [review]
Thanks for keeping the new changes in a separate commit for easier reviewing! Let's land this now, so that the next build at [4am UTC](https://github.com/mozilla-mobile/firefox-android/blob/main/.cron.yml#L13) will have this landed so we can monitor this in nightly during working hours.

Could you please land similar changes into Reference Browser as well (enabling minification too in our release types) so that we can catch future bugs as well? Thanks!

**Pull request merged**